packages/contract/src/DaimoFastCCTP.sol

```diff
@@ -0,0 +1,241 @@
+
+      // Record claimer as recipient. This ensures that nobody can mistakenly
+      // fastFinish() a transfer that has already been claimed.
+      handoffToRecipient[address(handoff)] = toAddr;
+    }
+
+    // If an LP fastFinish'd the transfer previously (recipient != toAddr),
+    // then the LP gains the difference fromAmount - toAmount (which can be 0).
+    toToken.transfer(recipient, fromAmount);
+  }
+}
+
+// This contract is deployed, then destroyed again in the same transaction.
+// CCTP lets us pass just a few  pieces of information from chain A to chain B:
+// sender, recipient, token, amount. This contract lets us encode all of the
+// FastCCTP send parameters into the sender address via CREATE2.
+contract EphemeralHandoff {
+  address payable private immutable _creator;
+  uint256 private immutable _fromAmount;
+  uint256 private immutable _toChainID;
+  IERC20 private immutable _toToken;
+
+  constructor(
+    uint256 /* fromChainID */,
+    address /* fromAddr */,
+    uint256 fromAmount,
+    uint256 toChainID,
+    address /* toAddr */,
+    uint256 /* toAmount */,
+    IERC20 toToken,
+    uint256 /* nonce */
+  ) {
+    _creator = payable(msg.sender);
+    _fromAmount = fromAmount;
+    _toChainID = toChainID;
+    _toToken = toToken;
+  }
+
+  modifier onlyCreator() {
+    require(msg.sender == _creator, "Only creator can call this function");
+    _;
+  }
+
+  function startTransferAndSelfDestruct(
+    ICCTPTokenMessenger cctpMessenger,
+    IERC20 token
+  ) public onlyCreator {
+    token.approve(address(cctpMessenger), _fromAmount);
+
+    // Send to CCTP. Recipient = this same EphemeralHandoff address on chain B.
+    uint32 destinationDomain = _chainIDToDomain(_toChainID);
+    bytes32 mintRecipient = bytes32(bytes20(address(this)));
+    cctpMessenger.depositForBurn(
+      _fromAmount,
+      destinationDomain,
+      mintRecipient,
+      address(token)
+    );
+
+    // Finally, self-destruct
+    destroy();
+  }
+
+  // Called immediately after the CCTP claim.
+  // Checks that the correct amount was received, then forwards to FastCCTP.
+  function receiveTransferAndSelfDestruct() public onlyCreator {
+    uint256 balanceReceived = _toToken.balanceOf(address(this));
+    require(balanceReceived == _fromAmount, "FCCTP: wrong balance received");
+
+    // Send to FastCCTP, which will forward to current recipient
+    _toToken.transfer(_creator, _fromAmount);
+
+    // Finally, self-destruct
+    destroy();
+  }
+
+  function destroy() public onlyCreator {
+    // This use of SELFDESTRUCT is compatible with EIP-6780.
+    // EphemeralHandoffs are deployed, then destroyed in the same transaction.
+    // solhint-disable-next-line
+    selfdestruct(_creator);
+  }
+
+  // This list only restricts destination chains = compatiable DaimoAccountV2
+  // home chains. The source chain could be any chain CCTP adds in the future.
+  function _chainIDToDomain(uint256 chainID) private pure returns (uint32) {
```

Comment:
nice

Reply:
should this mapping be a constructor arg?

Reply:
yea the daimoCCTPBridger currently takes in inputChainIds and outputDomains as constructor args --> maybe that would be better